lib/tables.c

```diff
@@ -2023,20 +2025,37 @@ table_sorter_sort_sites(table_sorter_t *self)
     /* Build the mapping from old site IDs to new site IDs and copy back into the table */
     ancestral_state_offset = 0;
     metadata_offset = 0;
+    jj = 0;
+    last_position = -1;
     for (j = 0; j < self->sites->num_rows; j++) {
-        self->site_id_map[sorted_sites[j].id] = (site_id_t) j;
-        self->sites->position[j] = sorted_sites[j].position;
-        self->sites->ancestral_state_offset[j] = ancestral_state_offset;
-        memcpy(self->sites->ancestral_state + ancestral_state_offset,
-            sorted_sites[j].ancestral_state,
-            sorted_sites[j].ancestral_state_length);
-        ancestral_state_offset += sorted_sites[j].ancestral_state_length;
-        self->sites->metadata_offset[j] = metadata_offset;
-        memcpy(self->sites->metadata + metadata_offset,
-            sorted_sites[j].metadata,
-            sorted_sites[j].metadata_length);
-        metadata_offset += sorted_sites[j].metadata_length;
-    }
+        self->site_id_map[sorted_sites[j].id] = jj;
```

Comment:
If I wrote this correctly, this means that there's a single "output" site ID for all "input" sites having the same position; so that all mutations referring to the same position will (afterwards) refer to the same site, even if they did not beforehand.

Reply:
Aha, yes; clever!